DotaStats/Modules/MatchesModule/Presenter/MatchesModulePresenter.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+
+protocol MatchesModuleInput: AnyObject {}
+
+protocol MatchesModuleOutput: AnyObject {
+    func matchesModule(_ module: MatchesModuleInput, didSelectPlayer matchId: Int)
+}
+
+final class MatchesModulePresenter {
+    weak var view: MatchesModuleViewInput?
+    private let matchesService: MatchesService
+    private var matches: [MatchCollectionPresenterData] = []
+    let output: MatchesModuleOutput
+    
+    required init(matchesService: MatchesService,
+                  output: MatchesModuleOutput)
+    {
+        self.matchesService = matchesService
+        self.output = output
+        self.state = .none
+    }
+    
+    private var state: MatchesModulePresenterState {
+        didSet {
+            oldValue.token?.cancel()
+            switch state {
+            case .result(let requestResult):
+                switch requestResult {
+                case .success(var matches):
+                    matches.sort { $0.startTime < $1.startTime }
+                    convertMatches(matches: matches)
+                    view?.updateState(matchesModuleState: MatchesModuleViewState.success)
```

Comment:
Тут тоже можно улучшить сигнатуру, например так.
```suggestion
                    view?.update(state: .success)
```

Даже если не улучшать сигнатуру, вот так полностью `MatchesModuleViewState.success` писать енамы не стоит, лучше опускать первую часть по возможности до `.success`